src/jit/simdcodegenxarch.cpp

```diff
@@ -671,7 +671,7 @@ void CodeGen::genSIMDScalarMove(
     var_types targetType, var_types baseType, regNumber targetReg, regNumber srcReg, SIMDScalarMoveType moveType)
 {
     assert(varTypeIsFloating(baseType));
-    if (compiler->getSIMDSupportLevel() == SIMD_AVX2_Supported)
+    if (compiler->getSIMDSupportLevel() >= SIMD_AVX_Supported)
```

Comment:
I don't believe we want to generate different code for the AVX case, to avoid multiplying our test matrix.

Reply:
But here is special. `vmovss` should use the sematics of "merge" (`vmovss xmm1, xmm1, xmm2`) rather than semtanc of "move`` (`vmovss xmm1, xmm2, xmm2`). Let me try to give a better solution.